workflow/sync/common.go

```diff
@@ -13,5 +13,10 @@ type semaphore interface {
 	getCurrentPending() []string
 	getName() string
 	getLimit() int
+	getLimitTimestamp() time.Time
+	resetLimitTimestamp()
 	resize(n int) bool
 }
+
+// expose for overriding in tests
+var now = time.Now
```

Comment:
This is clever, but I need to think about whether it's OK.

At the very least I'd prefer the variable had a longer name than just `now` so as to be more obvious as to what it does and to avoid anyone doing other bad things with it.

Reply:
yeah i didn't love the approach either i just couldn't come up with another way to mock time and didn't want to add a dependency

Reply:
made other fixes left this one unchanged for now since there's still a decision to be made if we even want to go this route

Reply:
I don't have a better way of doing this, but would still prefer the word `now` to be something more explicit. Perhaps just `nowFn` or `nowFunc` to illustrate it's an odd thing. Happy for other suggestions though.

Reply:
updated to `nowFn` i'm good with that approach